raft/src/main/java/org/apache/kafka/raft/RaftClient.java

```diff
@@ -172,15 +173,17 @@ default void beginShutdown() {}
      * uncommitted entries after observing an epoch change.
```

Comment:
It's probably worth adding a sentence or two about the new optimistic concurrency.